openvdb/points/PointMask.h

```diff
@@ -80,14 +83,51 @@ inline typename std::enable_if<std::is_same<typename MaskT::ValueType, bool>::va
     typename MaskT::Ptr>::type
 convertPointsToMask(const PointDataGridT& grid,
                     const openvdb::math::Transform& transform,
-                    const FilterT& filter = NullFilter());
+                    const FilterT& filter = NullFilter(),
+                    bool threaded = true);
+
+
+/// @brief No-op deformer (adheres to the deformer interface documented in PointMove.h)
+struct NullDeformer
+{
+    template <typename LeafT>
+    void reset(LeafT& leaf, size_t idx = 0) { }
+
+    template <typename IterT>
+    void apply(Vec3d&, IterT&) const { }
+};
+
+/// @brief Deformer Traits for optionally configuring deformers to be applied
+/// in index-space. The default is world-space.
+template <typename DeformerT>
+struct DeformerTraits
+{
+    static const bool IndexSpace = false;
+};
```

Comment:
Only bugbear with this is that you have to inject new specialisations into the openvdb::points namespace (supposing you are creating a new Deformer elsewhere),  whereas if this were a required member on the Deformer interface it wouldn't matter. 
For clarity suppose I am creating a new deformer FooDeformer: 
```
namespace foo {
struct FooDeformer(){
    template<LeafT>
    void reset(LeafT&, size_t) {}
    template<IterT>
    void apply(Vec3d&, IterT&) {}
}
} // namespace foo
```
then I also need to somewhere have 
```
namespace openvdb {
namespace points {
struct DeformerTraits <foo::FooDeformer>
{
    static const bool IndexSpace = true;
}
```
I'm not too well versed in whether this is a common approach and it probably doesn't matter that much but just pointing it out. Don't know whether there would be any better way to avoid this without requiring some inheritance or adding another required member into the Deformer?

Reply:
I agree it's not ideal. I thought quite carefully about this and my logic is as follows. World-space deformation is going to be by far the most common and should not require the user to have to define that they want a deformer to be world-space, it should be the default behaviour. To achieve maximum performance, this needs to be compile-time and cannot use virtual functions, so I didn't see any other way of exposing this that would allow an optional (but very rare) override to be able to make a deformer evaluate in index-space. Ultimately I kind of see this as a hook that I don't expect anyone to actually use (aside from a few unit tests), but it's there just in case someone wants to use it and means they don't have to patch the library in order to do this, which is far worse than injecting new specialisations into the openvdb::points namespace.